src/envoy/http/service_control/filter_stats.h

```diff
@@ -42,6 +77,15 @@ namespace service_control {
 struct ServiceControlFilterStats {
   ALL_SERVICE_CONTROL_FILTER_STATS(GENERATE_COUNTER_STRUCT,
                                    GENERATE_HISTOGRAM_STRUCT)
+  // Collect check call status.
+  static void collectCheckStatus(
+      ServiceControlFilterStats& filter_stats,
+      const ::google::protobuf::util::error::Code& code);
+
+  // Collect report call status.
+  static void collectReportStatus(
```

Comment:
How about this idea?  
1) we split ALL_SERVICE_CONTROL_FILTER_STATS into multiple list
SERVICE_CONROL_FILTER_STATS with old stats
SERVICE_CONTROL_X_STATS for all counters, x could be check, report and quota

3) struct ServiceControlFilterStats   {
   ALL_SERVICE_CONTROL_FILTER_STATS()
  SERVICE_CONTROL_CHECK_STATS
  SERVICE_CONTROL_REPORT_STATS
  SERVICE_CONTROL_QUOTA_STATS
}

Will this work?

Reply:
I have tried. It seems not work  when you call
{ALL_SERVICE_CONTROL_FILTER_STATS(
        POOL_COUNTER_PREFIX(scope, final_prefix),
        POOL_HISTOGRAM_PREFIX(scope, final_prefix)),
SERVICE_CONTROL_CHECK_STATS(
        POOL_COUNTER_PREFIX(scope, final_prefix),
        POOL_HISTOGRAM_PREFIX(scope, final_prefix)),
SERVICE_CONTROL_REPORT_STATS(
        POOL_COUNTER_PREFIX(scope, final_prefix),
        POOL_HISTOGRAM_PREFIX(scope, final_prefix)),
}

Reply:
I think it can be done.  Please try it again.  In the worse case, we create 4 struct for stats.  You can think of each of them is just one `struct xxxStats`.   This base class hold 4 of them.  It can be done.

The problem is we need to pass them into handler and client_cache.  So it is easier to wrap 4 of them into a new struct.

Reply:
Done